public/app/features/alerting/unified/NotificationPolicies.tsx

```diff
@@ -153,67 +188,57 @@ const AmRoutes = () => {
       rootRoute,
       insertPosition
     );
-    updateRouteTree(newRouteTree);
+    updateRouteTree.execute(newRouteTree);
   }
 
-  function updateRouteTree(routeTree: Route | RouteWithID) {
-    if (!result) {
+  // this function will make the HTTP request and tracks the state of the request
+  const [updateRouteTree, updateRouteTreeState] = useAsync(async (routeTree: Route | RouteWithID) => {
+    if (!rootRoute) {
       return;
     }
 
     // make sure we omit all IDs from our routes
     const newRouteTree = cleanRouteIDs(routeTree);
 
-    setUpdatingTree(true);
-
-    dispatch(
-      updateAlertManagerConfigAction({
-        newConfig: {
-          ...result,
-          alertmanager_config: {
-            ...result.alertmanager_config,
-            route: newRouteTree,
-          },
-        },
-        oldConfig: result,
-        alertManagerSourceName: selectedAlertmanager!,
-      })
-    )
-      .unwrap()
-      .then(() => {
-        appNotification.success('Updated notification policies');
-        if (selectedAlertmanager) {
-          refetchAlertGroups();
-        }
-        closeEditModal();
-        closeAddModal();
-        closeDeleteModal();
-      })
-      .finally(() => {
-        setUpdatingTree(false);
-      });
-  }
+    const newTree = await updateNotificationPolicyRoute({
+      newRoute: newRouteTree,
+      oldRoute: result,
+    });
+
+    appNotification.success('Updated notification policies');
```

Comment:
```suggestion
    !updateRouteTreeState.error && appNotification.success('Updated notification policies');
```

Reply:
I think we should check error before notifying that everything is fine right?☝️

Reply:
🤔 we shouldn't have to. Since we're in an `async` function, when `updateNotificationPolicyRoute` fails (throws) it will never get to execute this code unless it was successful.

Reply:
ohhh yes, you're right 👌